### DISCUSSION TITLE
Using the NewCSSMiddleware with echo

### DISCUSSION THREAD

**Discussion by dimmerz92** [a-h/templ#708] [Q&A] [answered]
I am using echo as my server. Just wondering if anyone has successfully managed to use NewCSSMiddleware with echo, and how they may have done it?

Otherwise, it's looking like I would need to define my own middleware that does the same thing.

**Comment by michaelByrne** [comment]
Pretty sure you're going to have to write your own middleware. Echo middleware uses Echo types. The templ middlware uses std lib http types. Shouldn't be hard to duplicate tho

**Comment by joerdav** [comment, marked as answer]
You should be able to leverage the `echo.WrapMiddleware`

If I convert our example in the docs:

```go
func main() {
  // Echo instance
  e := echo.New()

  // Middleware
  e.Use(middleware.Logger())
  e.Use(middleware.Recover())

  c1 := className()
  e.Use(echo.WrapMiddleware(func(h http.Handler) http.Handler {
    return NewCSSMiddleware(httpRoutes, c1)
  }))

  // Routes
  e.GET("/", hello)

  // Start server
  e.Logger.Fatal(e.Start(":1323"))
}
```

**Reply by dimmerz92** [reply]
That's awesome, thank you.
I didn't realise this was a method, I had a look all over the docs to check I didn't miss it, it doesn't seem to be in their docs.

**Reply by joerdav** [reply]
Fair, I just found it searching through godocs
